### PR TITLE
Download data

### PR DESCRIPTION
**Pull request by thomasyu888**

### CODE REVIEW

**Comment by thomasyu888** [comment]
https://github.com/Sage-Bionetworks/synapseDocs/issues/140

File: articles/downloading_data.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: "Downloading data"
+layout: article
+excerpt: This tutorial will cover the best practices of downloading a file, directory and managing where files are stored
```
Comment (kdaily) [inline articles/downloading_data.md]: missing period

File: articles/downloading_data.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+---
+
+
+## Downloading a file
+
+Every entity in synapse has a unique synapse Id associated with it.  It can be found on every entity page next to `Synapse ID:`, starting with `syn` ending with numbers (ie. `syn00123`)
```
Comment (kdaily) [inline articles/downloading_data.md]: Capitalize 'Synapse'

File: articles/downloading_data.md (hunk 1 above)
Comment (kdaily) [inline articles/downloading_data.md]: Managing where files are stored locally? Cuz my files are stored in the Synapse.

File: articles/downloading_data.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+{% tabs %}
+	{% tab Command %}
+		{% highlight bash %}
+synapse get syn00123
+		{% endhighlight %}
+	{% endtab %}
+
+	{% tab Python %}
+		{% highlight python %}
+Entity = syn.get("syn00123", downloadFile=False)
```
Comment (kdaily) [inline articles/downloading_data.md]: not common to capitalize variable names (only Class names) in Python.

File: articles/downloading_data.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+		{% endhighlight %}
+	{% endtab %}
+
+	{% tab R %}
+		{% highlight r %}
+Entity = synGet("syn00123", downloadFile= FALSE)
```
Comment (kdaily) [inline articles/downloading_data.md]: not common to capitalize variable names in R.

File: articles/downloading_data.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
+
+
+###Downloading File Entities
+`File` entities can be downloaded by using the `get` command. Without specifying the optional parameters, the file entity downloaded will always be the most recent version (See [versioning](http://docs.synapse.org/articles/versioning.html) for more details).
```
Comment (kdaily) [inline articles/downloading_data.md]: redundant with next section?

File: articles/downloading_data.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+{% tabs %}
+	{% tab Command %}
+		{% highlight bash %}
+synapse get syn00123 -v 1
+		{% endhighlight %}
+	{% endtab %}
+
+	{% tab Python %}
+		{% highlight python %}
+syn.get("syn00123", version=1)
```
Comment (kdaily) [inline articles/downloading_data.md]: inconsistent with previous example:

```
entity = syn.get("syn00123", version=1)
```

File: articles/downloading_data.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+		{% endhighlight %}
+	{% endtab %}
+
+	{% tab R %}
+		{% highlight r %}
+synGet("syn00123", version=1)
```
Comment (kdaily) [inline articles/downloading_data.md]: inconsistent with previous example:

```
entity <- synGet("syn00123", version=1)
```

File: articles/downloading_data.md (hunk 4 above)
Comment (kdaily) [inline articles/downloading_data.md]: `<-` not `=`

```
entity <- synGet("syn00123", downloadFile=FALSE)
```

File: articles/downloading_data.md (hunk 3 above)
Comment (kdaily) [inline articles/downloading_data.md]: The section is called 'Downloading a File' - why is `downloadFile=False` set?

File: articles/downloading_data.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
+
+
+####Download Location
+By default the download location will always be the synapse cache directory, except for the command line client where the default download location is your current working directory.
```
Comment (kdaily) [inline articles/downloading_data.md]: And the Synapse cache directory is where, and/or defined how?

File: articles/downloading_data.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+{% tabs %}
+	{% tab Command %}
+		{% highlight bash %}
+synapse get syn00123 --downloadLocation ./path/to/folder
+		{% endhighlight %}
+	{% endtab %}
+
+	{% tab Python %}
+		{% highlight python %}
+syn.get("syn00123", downloadLocation="./path/to/folder")
```
Comment (kdaily) [inline articles/downloading_data.md]: Inconsistent with prev example (see above)

File: articles/downloading_data.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+		{% endhighlight %}
+	{% endtab %}
+
+	{% tab R %}
+		{% highlight r %}
+synGet("syn00123", downloadLocation="./path/to/folder")
```
Comment (kdaily) [inline articles/downloading_data.md]: Inconsistent with prev example (see above)

File: articles/downloading_data.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
+
+
+####Command line download
+The synapse command line client offers two ways of downloading synapse files that the other clients don't.  First, it allows for recursive downloading of files, maintaining the folder structure that is present on syanpse.  Second, users can pass in a query statement (See [here](http://docs.synapse.org/articles/annotation_and_query.html#queries) to learn more about query statements) to download all the files that is returned from that query statement.
```
Comment (kdaily) [inline articles/downloading_data.md]: Don't use contractions in formal docs. Capitalize Synapse. Check tense of last sentence.

File: articles/downloading_data.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+	{% tab R %}
+		{% highlight r %}
+synGet("syn00123", downloadLocation="./path/to/folder")
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
+
+
+####Command line download
```
Comment (kdaily) [inline articles/downloading_data.md]: inconsistent Capitalization of Header (doesn't follow others that Capitalize All Words)

File: articles/downloading_data.md [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+```
+# syn00123 has to be a folder or a project
+syanpse get -r syn00123
+# In the case below, syn00123 has to be a project.  Any query statement would work. 
+synapse get -q 'select id from entity where projectId == "syn00123"'
+```
+
+####Download Tables
+
+Please view [here](http://docs.synapse.org/articles/tables.html#making-changes-to-tables) to learn how to use table entities.
```
Comment (kdaily) [inline articles/downloading_data.md]: `Table` entities

File: articles/downloading_data.md [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+
+
+####Download Wikis
+The structure of a project's wikipage can be extracted through the `R` and `python` client.  The id, title and parent wikipage of each subwikipage is also determined through the same method.
```
Comment (kdaily) [inline articles/downloading_data.md]: `Project`s `Wiki` page

File: articles/downloading_data.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+		{% highlight r %}
+entity <- synGet("syn00123")
+synGetWikiHeaders(entity)
+		{%endhighlight %}
+	{% endtab %}
+
+{% endtabs %}
+
+
+The markdown and other information of a project subwikipage can be obtained by knowing the id of the wiki id.  The wikipage id can either be obtained through the above method or can be found in the url: "www.synapse.org/#!Synapse:syn00123/wiki/12345" where 12345 is the wikipage id.
```
Comment (kdaily) [inline articles/downloading_data.md]: `Project` sub-`Wiki` page; `Wiki` ID.

**Pull request merged**